### ISSUE TITLE
kubectl: Pod with deletionTimestamp shown as Terminating after Failed/Succeeded

### ISSUE THREAD

**Issue by alculquicondor** [kubernetes/kubernetes#120207]
### What happened?

A pod with a finalizer and a deletion timestamp is listed as "Terminating" in kubectl get and describe, even after the phase is Failed/Succeeded

### What did you expect to happen?

The phase Failed or Succeeded to be shown

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a long running pod with a finalizer
2. Issue a pod delete
3. Use `kubectl get -o wide` or `kubectl describe` to see the pod.

### Anything else we need to know?

This causes confusion, as the Pod doesn't account for any resource usage when its Failed or Succeeded.
This should be backported to all supported versions.

### Kubernetes version

<details>

master

</details>

### Cloud provider

<details>
N/A
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by alculquicondor** [comment]
any progress on this?

**Comment by lowang-bh** [comment]
Hi, @alculquicondor , one thing to be comfirm: just need to change the `Terminating` to what as `phase` shows when use kubect get pod if pod has deletionTimestamp, right?
<img width="843" alt="image" src="https://github.com/kubernetes/kubernetes/assets/21003791/8a1da44e-9b22-41ab-8c9a-0803e2c09ca3">

**Comment by alculquicondor** [comment]
Yes, I would say Succeeded or Failed. Terminating only makes sense if the Pod is in phase Running and has the deletionTimestamp.
Although I don't know if anyone in SIG CLI disagrees. Worth confirming.
